Sources/MessagingInApp/State/InAppMessageManager.swift

```diff
@@ -53,6 +53,10 @@ public class InAppMessageManager {
         }
     }
 
+    func dispatchAsync(action: InAppMessageAction) async {
```

Comment:
added this cause it was very helpful in streamlining tests, otherwise we would need to use expectations and timeouts.

Reply:
Both approached creates and awaits a `Task` object, adding unnecessary overhead. Since we are already calling it from `async` method we don't need to create a Task and then await it. 

I have kept the same flow but refined it a bit more.

Reply:
Since this is only used by tests, I'm not sure what unnecessary overhead is being added here. My intention was to avoid including `dispatchAsync` in production code since we don't need it there. I'm okay with even making `store` internal and adding this extension just for tests.

Also, can you please revert the recent change? Calling `dispatchAsync` from `dispatch` is completely unnecessary in my opinion. As all our production code uses `dispatch`, and now we've added an extra function call to `dispatchAsync` just for tests, which doesn't seem needed to me.

Reply:
Okay there are a couple of points here, let me answer one by one

> I'm not sure what unnecessary overhead is being added here

```
extension InAppMessageManager { 
func dispatchAsync(action: InAppMessageAction) async { 
await dispatch(action: action).value } 
}
``` 

This method, can directly get value from `store.dispatch`, it doesn't need a `task`, but because we are using `dispatch(action: action).value ` we are unnecessarily creating Task and getting the value. 

> My intention was to avoid including dispatchAsync in production code

I don't understand, why? what are your reservations about it? The reason we created `tasks with callback` was so we don't have to add `async/await` everywhere. Internally we are still using actors and creating Tasks and awaiting. 

So what we have done is create two methods:
- Async method that can be called from any async method
- Method which creates a `Task` and calls the Async method internally.

It's the same thing we have done in `data pipeline` classes, where we would expose different options but keep one primary option.

If you exclude the "just for tests" part, why is this not a good addition and what overhead would it be creating? We are literally doing the same thing but now we just have exposes a method to do it. 

- It provides consistency as all dispatch calls go through a single async implementation, ensuring consistent behavior across your app.
- It reduces code duplication, as the non-async version now calls the async version.
- If you need to modify the dispatch logic in the future, you only need to update one place (the async version).

Reply:
If anything is needed in the future, we can address it then and figure out how to improve. We don't need to make assumptions and modify our code unnecessarily.

I don’t think this requires a lengthy discussion, but here are a few important points:

- I'm okay with adding `dispatchAsync` if you're adamant about keeping it, but I'm strongly against adding any code to production that we know for sure is completely unused.

- We don't need to change `dispatch` here, and we shouldn’t unless it's absolutely necessary. It should call the store directly and calling `dispatchAsync` is unnecessary from `dispatch` IMO. I'm fine keeping `dispatchAsync` if you can revert `dispatch` to the previous implementation.

Reply:
This is currently, even if it's the use-case is `test` for now, next up it could very much be using swift concurrency through out the SDK. 

It doesn't make sense to me why we would have 2 methods doing the same thing that is calling `store.dispatch` but separately. It just means, we are saying, whenever we need to make a change in logic, we update two methods for it. 

We are not `changing` dispatch here, we are actively calling another method which would be the primary method for `dispatching.` There is ZERO logic change.